packages/main/src/plugin/extensions-updater/extensions-updater.ts

```diff
@@ -137,8 +139,26 @@ export class ExtensionsUpdater {
 
         // if found compare versions
         const installedVersion = installedExtension.version;
+        const appVersion = app.getVersion();
+
+        // coerce the podman desktop Version
+        const currentPodmanDesktopVersion = coerce(appVersion);
+
+        // filter out versions non-compliant with this version of Podman Desktop
+        const availableVersions = availableExtension.versions.filter(version => {
+          const extensionRequirePodmanDesktopVersion = version.podmanDesktopVersion;
+          if (extensionRequirePodmanDesktopVersion && currentPodmanDesktopVersion) {
+            //  keep the versions that are compatible with this version of Podman Desktop
+            return satisfies(currentPodmanDesktopVersion, extensionRequirePodmanDesktopVersion);
+          } else {
+            // if no version is specified, keep the version
+            return true;
+          }
+        });
+
+        const filteredPreviewVersions = availableVersions.filter(version => version.preview === false);
         // take latest version
-        const latestAvailableVersion = availableExtension.versions.filter(version => version.preview === false)?.[0];
+        const latestAvailableVersion = filteredPreviewVersions?.[0];
```

Comment:
I was thinking if they uses the `Stop remind me` option, then we have nothing telling them to update nicely. But even if they do have it, showing only compatible latest seem counter productive, as we probably do not want to promote older version

Reply:
Moreover, user could be misleaded, they would think they use the latest version, when they are using the latest version they are compatible with

Reply:
why would they be misleaded. the version that they're using is being displayed

Also it needs to be thought in a different way:
if we release a new AI Lab version, it will be updated on all Podman Desktop instances and it won't work if there are some API changes
You want to have automatic updates only if it will work.

Reply:
Oh I don't want automatic update ! Sorry for the confusion, i forget we automatically update it, i thought we were simply displaying the latest version, and update on user action

Reply:
here it's the code for the automatic update